src/main/kotlin/entry/dsm/gitauth/equusgithubauth/domain/auth/service/GithubUserTokenValidationService.kt

```diff
@@ -0,0 +1,57 @@
+package entry.dsm.gitauth.equusgithubauth.domain.auth.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpStatus
+import org.springframework.http.RequestEntity
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Service
+import org.springframework.web.client.RestTemplate
+
+@Service
+class GithubUserTokenValidationService(
+    private val restTemplate: RestTemplate
+) {
+    private val logger = LoggerFactory.getLogger(GithubUserTokenValidationService::class.java)
+
+    fun validateAccessToken(token: String) {
+        token.takeIf { it.isNotBlank() } ?: throw IllegalArgumentException("Access token is empty.")
+        val isTokenValid = retry(3, 2000) { isTokenActive(token) } // Add delay between retries
+        if (!isTokenValid!!) {
+            throw IllegalArgumentException("Access token is expired or invalid.")
+        }
+    }
+
+    private fun retry(times: Int, delay: Long = 0L, block: () -> Boolean?): Boolean? {
+        repeat(times - 1) {
+            val result = try {
+                block()
+            } catch (e: Exception) {
+                logger.error("Retry failed: ${e.message}")
+                null
+            }
+            if (result == true) return result
+            Thread.sleep(delay)
+        }
+        return block()
+    }
```

Comment:
```suggestion
 private fun retry(times: Int, delay: Long = 0L, block: () -> Boolean): Boolean {
    repeat(times - 1) {
        val result = try {
            block()
        } catch (e: Exception) {
            logger.error("Retry failed: ${e.message}")
            false
        }
        if (result) return true
        Thread.sleep(delay)
    }
    return block()
}
```
 nullable 타입(Boolean?)을 반환을 해 호출한 쪽에서 !!로 강제 언래핑하고 있어  block()이 null을 반환할 가능성이 있는 것 같아 예상하지 못한 부분에서 NullPointerException이 발생할 것 같습니다
 제가 작성한 로직으로 변경하는것은 어떤가요?

Reply:
e900a217250feb5f72be95cd6d5d573d1e77ff72

restTemplate -> Github FeignClient를 이용하도록 변경했습니다!